### ISSUE TITLE
[rllib] DDPG TWIN_Q LOSS ERROR

### ISSUE THREAD

**Issue by mvindiola1** [ray-project/ray#10653]
The TD3 loss according to OpenAI Spinning up is:
![L1](https://spinningup.openai.com/en/latest/_images/math/7d5c18f49a242cc3eec554f717fe4f3bfc119bab.svg)

![L2](https://spinningup.openai.com/en/latest/_images/math/cd73726a8a3845ade467aed57714912f868f6b36.svg)

They implement this as you would expect:
````
# MSE loss against Bellman backup
loss_q1 = ((q1 - backup)**2).mean()
loss_q2 = ((q2 - backup)**2).mean()
loss_q = loss_q1 + loss_q2
````

The rllib implementation is off and I ran into training issues when using it.

Tensorflow:
https://github.com/ray-project/ray/blob/39c598bab0929f007a49daa7262a7c9ee6492221/rllib/agents/ddpg/ddpg_tf_policy.py#L174-L184

Pytorch:
https://github.com/ray-project/ray/blob/5851e893eee491e569fe1421314172bd5b37d583/rllib/agents/ddpg/ddpg_torch_policy.py#L124-L134

The error is on line 177 and 127 respectively where the td_error and twin_td_error are added together and then squared a couple lines later. Removing that line would bring the implementation in line with the spinningup reference implementation. This worked for me to improve training on my problem. They also do not multiply by the constant (0.5) but I would not expect that to make much of a difference either way.

**Comment by ericl** [comment]
Interesting, do you want to push a fix. Also cc @michaelzhiluo if you could take a look here.
